### ISSUE TITLE
Call Depth Overflow During Publish

### ISSUE THREAD

**Issue by sl304227-nh** [SQLPlayer/azure.datafactory.tools#206]
**Describe the bug**
In v0.96.0, when publishing an ADF via `Publish-AdfV2FromJson`, the deleting of a `LinkedService` causes a overflow crash.

**To Reproduce**
1) Create component in an existing ADF which has a duplicate id / name using the Azure ADF UI.
2) Export the definition of the entire ADF via Git.
3) Confirm the duplicate component throws a warning via `Test-AdfCode`.
4) Try to publish to the same ADF via `Publish-AdfV2FromJson`, while removing both instances of the duplicate component using `DeleteNotInSource`:
```
$Option = New-AdfPublishOption;
$Option.DeleteNotInSource = $true;
$Option.DoNotDeleteExcludedObjects = $false;

Publish-AdfV2FromJson `
    -RootFolder $RootFolder `
    -ResourceGroupName $ResourceGroupName `
    -DataFactoryName $DataFactoryName `
    -Stage $DeployEnv `
    -Location $Location `
    -Method $Method `
    -Option $Option;
5) Observe that the unused resources in the ADF will be deleted successfully, except once the duplicate component is hit.  The duplicate component is the one that will cause the overflow error.

```
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Removing object: [LinkedService].[bdas_dw]
Remove-AdfObject: /usr/local/share/powershell/Modules/azure.datafactory.tools/0.96.0/private/Remove-AdfObject.ps1:113
Line |
 113 |              Remove-AdfObject -adfSource $adfSource -obj $_ -adfInstan …
     |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The script failed due to call depth overflow.

##[error]PowerShell exited with code '1'.

```

I inherited this existing ADF on a project.  When I exported it to Git and ran the `Test-AdfCode` I noticed the Warning.  Lo and behold, the overflow is on the same component that threw the warning.

**Comment by druzin** [comment]
I'm seeing the same behavior.

**Comment by druzin** [comment]
> Check whether you have another object in ADF called `bdas_dw`?

Ah - on my end - I have two objects in ADF with the same name - and I'm trying to delete both of them. Would that explain it ?

**Comment by sl304227-nh** [comment]
Yes, there is a duplicate in the ADF, which is allowed by Azure, but this tool cannot delete it.  It is a defect in this tool IMO.  It should handle the remote deleting of duplicates gracefully, and not crash.

**Comment by sl304227-nh** [comment]
I don't know why Azure allows duplicates, but it does.

**Comment by sl304227-nh** [comment]
> > Check whether you have another object in ADF called `bdas_dw`?
> 
> Ah - on my end - I have two objects in ADF with the same name - and I'm trying to delete both of them. Would that explain it ?

yes, this tool can't handle that scenario apparently
